### PR TITLE
[libxml2] add more features

### PR DESCRIPTION
**Pull request by yurybura**
Move hardcoded external dependencies (libiconv, liblzma, zlib) to features, disable legacy flag by default.

### CODE REVIEW

**Review by dg0yt — CHANGES_REQUESTED** [review]
Please don't change the formatting.

The network features are controversial.
FTP is off by default upstream, so it should be off by default in vcpkg.
HTTP is on by default upstream, but this breaks uwp.
IMO both could be off by default because they are don't offer secure communication.

**Comment by yurybura** [comment]
> Please don't change the formatting.

Ok.

> The network features are controversial. FTP is off by default upstream, so it should be off by default in vcpkg. 

FTP is off by default in the library, but it was enabled by default in VCPKG. Ok, I will follow library defaults.

> HTTP is on by default upstream, but this breaks uwp.

HTTP and FTP are off on UWP in port's JSON: https://github.com/microsoft/vcpkg/pull/30154/files#diff-642422f43590726b42670763cb558af55056b3c1b9e8218f00a01bf1d753c102R32

**Comment by dg0yt** [comment]
> HTTP and FTP are off on UWP in port's JSON:

They are not supported on UWP but still default features on all platforms. That's why the default installation is now broken for UWP.

**Comment by yurybura** [comment]
> > HTTP and FTP are off on UWP in port's JSON:
> 
> They are not supported on UWP but still default features on all platforms. That's why the default installation is now broken for UWP.

@dg0yt Thank you for the help. Can you explain how to find this error in CI log? I see only failing ffmpeg, but not libxml2.
I thought that not supported features shouldn't be enabled by default.

**Comment by dg0yt** [comment]
> Can you explain how to find this error in CI log? I see only failing ffmpeg, but not libxml2.
> I thought that not supported features shouldn't be enabled by default.

The last assumption is wrong, unfortunately.

In CI, you can find this:
~~~
2023-03-13T21:24:25.0719853Z Determining parent hashes using HEAD~1
...
2023-03-13T21:24:37.7602418Z                          libxml2:x64-uwp:        *: fc1251a26227be5ea7e8d85511264909ea6130594175934b5e57573f149dd7c7
...
2023-03-13T21:24:38.2563362Z The following packages will be built and installed:
...
2023-03-13T21:24:38.5708655Z     libxml2[core]:x64-uwp -> 2.10.3
~~~
i.e. it tinyxml2 would build for master, but now for your PR merged into master:
~~~
2023-03-13T21:24:38.8263790Z Running CI using parent hashes
2023-03-13T21:24:38.9107487Z HEAD is now at 406598af0 Merge 8b1c3a90708f580b7c849e60c3f03a5119114251 into f09a2226865dd00a54b983933dd29ab29bca8bf6
...
2023-03-13T21:24:50.6519768Z                          libxml2:x64-uwp:  cascade
~~~
i.e. libxml2 is already known to be unsupported and is not even considered for build, which in turn causes
~~~
2023-03-13T21:24:51.4587144Z ##[error]REGRESSIONS:
REGRESSION: vcpkg-ci-ffmpeg:x64-uwp cascaded, but it is required to pass. (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
~~~
because is vcpkg-ci-ffmpeg is required to pass, but has unsatisfied requirements.

**Comment by dg0yt** [comment]
You can probably also check locally with  `vcpkg install --dry-run libxml2:x64-uwp`.

**Comment by jimwang118** [comment]
@yurybura There is a conflict in the code you submitted, and the conflict needs to be resolved before CI can run.

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/libxml2/vcpkg.json [hunk 1]
```diff
@@ -1,25 +1,72 @@
...
+          "name": "libxml2",
+          "default-features": false,
+          "features": [
+            "ftp",
+            "http"
+          ],
+          "platform": "!uwp"
+        }
+      ]
+    },
```
[the remark this answers is not in the document]
Comment (yurybura) [reply ports/libxml2/vcpkg.json]: Reverted.

**Comment by yurybura** [comment]
@jimwang118 All checks has been done.

**Comment by jimwang118** [comment]
All feature passed with following triplets:

x86-windows
x64-windows
x64-windows-static

**Review by LilyWangLL — APPROVED** [review]

**Pull request merged**